app/controllers/starter_file_groups_controller.rb

```diff
@@ -6,7 +6,11 @@ class StarterFileGroupsController < ApplicationController
 
   def create
     assignment = Assignment.find_by(id: params[:assignment_id])
-    assignment.starter_file_groups.create(update_params)
+
+    new_params = update_params
```

Comment:
Do you think it makes more sense to put this logic in a `before_validate` callback in the StarterFileGroup model?

Something like:

```ruby
# app/models/starter_file_group.rb

before_validate :set_name, on: :create

private

def set_name
    return if self.name || self.assignment.nil?

    current_names = self.assignment.starter_file_groups.pluck(:name).to_set
    new_name = I18n.t('assignments.starter_file.new_starter_file_group')
    i = 1
    while current_names.include? new_name
      new_name = "#{I18n.t('assignments.starter_file.new_starter_file_group')} (#{i})"
      i += 1
    end
    self.name = new_name
end
```

That way all the logic is contained in the model (so that it will be enforced no matter where it is created) and you don't have to worry about modifying the params in the controller(s).

Reply:
Yeah that's a good idea, thanks. I'll make the change soon.